### DISCUSSION TITLE
how to get nx to execute a chain of pre scripts

### DISCUSSION THREAD

**Discussion by dagda1** [nrwl/nx#7744] [General]
I am working on a monorepo that makes liberal use of npm's `pre` and `post` scripts.

```json
    "pretsc": "yarn generate",
    "tsc": "tsc",
    "prebuild": "yarn tsc",
    "build": "lerna run build",
```

running `yarn nx run-many --target=build --all --parallel` means these pre scripts do not get built.

I know there is a `dependsOn` property but I am struggling to find a good example of this working.

Is there a link to an actual real repo with an up to date version that shows this working?

**Comment by brandonroberts** [comment]
The pre/post scripts should work if you are just using the `projects` property in the `workspace.json` to reference the path to that project. Let's say you have a monorepo with Lerna currently. Running

```sh
npx add-nx-to-monorepo
```

Will add Nx and a bare minimum configuration for your existing projects. Running `nx build my-project` defers to the npm scripts, which may have their own pre/post scripts.

**Comment by dagda1** [comment]
@brandonroberts these commands actually exist in the root and I am looking to replace `lerna`.

I think this is actually down to yarn berry that does not call `pre` and `post` scripts.

Is it possible to use globs with nx?

e.g. in lerna I could do

`lerna run --scope=@devex/backend-*`

**Comment by brandonroberts** [comment]
Ahh, ok. Nx doesn't support globs yet, but affected commands should be more efficient in knowing what's changed, and what to run against those changes. If you wanted to do it manually, you could programmatically have a script that grabs projects based on a glob and pass those projects to `nx run-many --projects={globbedProjects}`

**Comment by dagda1** [comment]
@brandonroberts many thanks for the answers

**Comment by dagda1** [comment]
@brandonroberts actually nx might negate my need for a glob.

If I run

`yarn nx run-many --target=generate --all --parallel`

Will it avoid packages without a `generate` script?

**Reply by brandonroberts** [reply]
Yes, it will ignore projects that don't have the provided `target` or script in this case
